pkg/cmd/registry/artifact/crud/list/list.go

```diff
@@ -0,0 +1,210 @@
+		Example: `
+## List all artifacts for the default artifacts group
+rhoas service-registry artifact list
+
+## List all artifacts with explicit group 
+rhoas service-registry artifact list --group=my-group
+
+## List all artifacts with limit and group
+rhoas service-registry artifact list --page=2 --limit=10
+		`,
+		Args: cobra.RangeArgs(0, 1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if opts.outputFormat != "" && !flagutil.IsValidInput(opts.outputFormat, flagutil.ValidOutputFormats...) {
+				return flag.InvalidValueError("output", opts.outputFormat, flagutil.ValidOutputFormats...)
+			}
+
+			if len(args) > 0 {
+				opts.group = args[0]
+			}
+
+			if opts.page < 1 || opts.limit < 1 {
+				return errors.New("page and limit values should be bigger than 1")
+			}
+
+			if opts.registryID != "" {
+				return runList(opts)
+			}
+
+			cfg, err := opts.Config.Load()
+			if err != nil {
+				return err
+			}
+
+			if !cfg.HasServiceRegistry() {
+				return errors.New("No service Registry selected. Use rhoas registry use to select your registry")
+			}
+
+			opts.registryID = fmt.Sprint(cfg.Services.ServiceRegistry.InstanceID)
+
+			return runList(opts)
+		},
+	}
+
+	cmd.Flags().StringVarP(&opts.group, "group", "g", "", "Artifact group")
+
+	cmd.Flags().Int32VarP(&opts.page, "page", "", 1, "Page number")
+	cmd.Flags().Int32VarP(&opts.limit, "limit", "", 100, "Page limit")
+
+	cmd.Flags().StringVarP(&opts.registryID, "instance-id", "", "", "Id of the registry to be used. By default uses currently selected registry")
+	cmd.Flags().StringVarP(&opts.outputFormat, "output", "o", "", "Output format (json, yaml, yml, table)")
+
+	flagutil.EnableOutputFlagCompletion(cmd)
+
+	return cmd
+}
+
+func runList(opts *Options) error {
+	logger, err := opts.Logger()
+	if err != nil {
+		return err
+	}
+
+	if opts.group == "" {
+		logger.Info("Group was not specified. Using " + util.DefaultArtifactGroup + " artifacts group.")
+		opts.group = util.DefaultArtifactGroup
+	}
+
+	connection, err := opts.Connection(connection.DefaultConfigRequireMasAuth)
+	if err != nil {
+		return err
+	}
+
+	api := connection.API()
+
+	a, _, err := api.ServiceRegistryInstance(opts.registryID)
+	if err != nil {
+		return err
+	}
+	request := a.ArtifactsApi.ListArtifactsInGroup(context.Background(), opts.group)
+
+	request = request.Offset((opts.page - 1) * opts.limit)
+	request = request.Limit(opts.limit)
+	request = request.Orderby(registryinstanceclient.SORTBY_CREATED_ON)
+	request = request.Order(registryinstanceclient.SORTORDER_ASC)
+
+	response, _, err := request.Execute()
+	if err != nil {
+		return registryinstanceerror.TransformError(err)
+	}
+
+	totalCount := opts.page * opts.limit
+	if len(response.Artifacts) != 0 && response.GetCount() < totalCount {
+		logger.Info("Provided limit and page arguments are larger than total count of elements on the server", response.GetCount())
+		return nil
+	}
```

Comment:
I'm not entirely sure why this is needed?

page=1
limit=10

Therefore you will get the first 10 possible items.

totalCount = 10

if the number of artifacts is less than 10, then log this message and return nil?

Reply:
Actually we can remove it